### PR TITLE
Add a solution for Homework 5 Chapter 1

### PR DESCRIPTION
**Pull request by tzurowski**

### CODE REVIEW

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 1]
```diff
@@ -88,8 +142,42 @@ public static int[] InsertLast(int[] array, int number)
...
+        public static int[] InsertItemToIndex(int[] array, int number, int index)
+        {
+
+            if (array == null || array.Length == 0)
+            {
+                if (array != null && array.Length == 0 && index != 0)
+                {
+                    return array;
+                }
+                int[] newArrayWithOneItem = new int[1] { number };
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Remove the array size specification; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134&bid=16588760)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 2]
```diff
@@ -88,8 +142,42 @@ public static int[] InsertLast(int[] array, int number)
...
+                return array;
+            }
+            int[] newArray = new int[arrayLength + 1];
+            int newArrayLength = newArray.Length;
+            for (int i = 0, j = 0; i < newArrayLength; i++)
+            {
+                if (i == index)
+                {
+                    newArray[i] = number;
+                    j = 1;
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Do not update the loop counter 'j' within the loop body.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134&bid=16588760)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 3]
```diff
@@ -88,8 +143,43 @@ public static int[] InsertLast(int[] array, int number)
...
+        public static int[] InsertItemToIndex(int[] array, int number, int index)
+        {
+
+            if (array == null || array.Length == 0)
+            {
+                if (array != null && array.Length == 0 && index != 0)
+                {
+                    return array;
+                }
+                int[] newArrayWithOneItem = new int[] { number };
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134&bid=16588760)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 4]
```diff
@@ -51,8 +80,31 @@ public static int[] RemoveFirst(int[] array)
...
+            for (int i = 0; i < arrayLength; i++)
+            {
+                if (i == index)
+                {
+                    removedIndex = 1;
+                }
+                newArray[i] = array[i + removedIndex];
+            }
+            return newArray;
         }
```
Comment (ellmaster) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: This is identical to what RemoveAt is supposed to do. Put it in there.

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 5]
```diff
@@ -88,8 +143,43 @@ public static int[] InsertLast(int[] array, int number)
...
+                    newArray[i] = number;
+                    additionalIndex = 1;
+                }
+                else
+                {
+                    newArray[i] = array[i - additionalIndex];
+                }
+            }
+            return newArray;
         }
```
Comment (ellmaster) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Same here.

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 4 above)
Comment (tzurowski) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: So I can't use one method to RemoveAt, RemoveLast and RemoveFirst? I should use almost the same code in all three method?

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 5 above)
Comment (tzurowski) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: Corrected, thanks

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5036134&bid=16588760)

**Pull request closed without merging**